internal/state/graph.go

```diff
@@ -123,46 +178,75 @@ func bindHTTPRouteToListeners(
+		// FIXME(pleshakov) Support empty section name
+		if p.SectionName == nil || *p.SectionName == "" {
 			continue
 		}
 
-		ignored = false
+		// if the namespace is missing, assume the namespace of the HTTPRoute
+		ns := ghr.Namespace
+		if p.Namespace != nil {
+			ns = string(*p.Namespace)
+		}
 
 		name := string(*p.SectionName)
 
-		// Step 2 - Find a listener
-
-		// FIXME(pleshakov)
-		// For now, let's do simple matching.
-		// However, we need to also support wildcard matching.
-		// More over, we need to handle cases when a Route host matches multiple HTTP listeners on the same port when
-		// sectionName is empty and only choose one listener.
-		// For example:
-		// - Route with host foo.example.com;
-		// - listener 1 for port 80 with hostname foo.example.com
-		// - listener 2 for port 80 with hostname *.example.com;
-		// In this case, the Route host foo.example.com should choose listener 2, as it is a more specific match.
-
-		l, exists := listeners[name]
-		if !exists {
-			r.InvalidSectionNameRefs[name] = struct{}{}
+		// Below we will figure out what Gateway resource the parentRef references and act accordingly. There are 3 cases.
+
+		// Case 1: the parentRef references the winning Gateway.
+
+		if gw != nil && gw.Namespace == ns && gw.Name == string(p.Name) {
+
+			// Find a listener
+
+			// FIXME(pleshakov)
+			// For now, let's do simple matching.
+			// However, we need to also support wildcard matching.
+			// More over, we need to handle cases when a Route host matches multiple HTTP listeners on the same port when
+			// sectionName is empty and only choose one listener.
+			// For example:
+			// - Route with host foo.example.com;
+			// - listener 1 for port 80 with hostname foo.example.com
+			// - listener 2 for port 80 with hostname *.example.com;
+			// In this case, the Route host foo.example.com should choose listener 2, as it is a more specific match.
+
+			l, exists := listeners[name]
+			if !exists {
+				r.InvalidSectionNameRefs[name] = struct{}{}
+				ignored = false
+				continue
+			}
+
+			accepted := findAcceptedHostnames(l.Source.Hostname, ghr.Spec.Hostnames)
+
+			if len(accepted) > 0 {
+				for _, h := range accepted {
+					l.AcceptedHostnames[h] = struct{}{}
+				}
+				r.ValidSectionNameRefs[name] = struct{}{}
+				l.Routes[getNamespacedName(ghr)] = r
+			} else {
+				r.InvalidSectionNameRefs[name] = struct{}{}
+			}
+
+			ignored = false
 			continue
 		}
 
-		accepted := findAcceptedHostnames(l.Source.Hostname, ghr.Spec.Hostnames)
+		// Case 2: the parentRef references an ignored Gateway resource.
 
-		if len(accepted) > 0 {
-			for _, h := range accepted {
-				l.AcceptedHostnames[h] = struct{}{}
-			}
-			r.ValidSectionNameRefs[name] = struct{}{}
-			l.Routes[getNamespacedName(ghr)] = r
-		} else {
+		key := types.NamespacedName{Namespace: ns, Name: string(p.Name)}
+
+		if _, exist := ignoredGws[key]; exist {
 			r.InvalidSectionNameRefs[name] = struct{}{}
+
+			ignored = false
```

Comment:
shouldn't ignore be true here?

Reply:
in this case, we process an HTTPRoute but do not bind to any listener, because the HTTPRoute references an ignored Gateway resource. That Gateway resource still belongs to NGINX Gateway though. Thats why ignore is false for the HTTPRoute

Reply:
agree, it was confusing. hope it is less confusing now.